### ISSUE TITLE
require-jsdoc failing when preceded by static object declaration

### ISSUE THREAD

**Issue by virtuacoplenny** [eslint/eslint#8678]
This wasn't happening for us yesterday morning but I now see builds from last night failing with require-jsdoc.

Sample repro: https://github.com/virtuacoplenny/eslint-missing-jsdoc

**Tell us about your environment**

* **ESLint Version:**3.19.0
* **Node Version:**7.7.2
* **npm Version:**4.1.2

**What parser (default, Babel-ESLint, etc.) are you using?**babel-eslint 7.2.3

**Please show your full configuration:**

```
module.exports = {
    'parser': 'babel-eslint',
    'plugins': [
        'react'
    ],
    'rules': {
        'require-jsdoc': [
            'error',
            {
                'require': {
                    'MethodDefinition': true
                }
            }
        ],
    }
};
```

**What did you do? Please include the actual source code causing the issue.**

```js
import React, { Component } from 'react';

/**
 * Sample jsdoc.
 */
class Foo extends Component {
    /**
     * Sample jsdoc.
     */
    static propTypes = {
        bar: React.PropTypes.bool
    }

    /**
     * Sample jsdoc.
     *
     * @returns {ReactElement}
     */
    render() {
        return (
            <div>{ this.props.bar }</div>
        );
    }
}
```

**What did you expect to happen?** For require-jsdoc to pass.

**What actually happened? Please include the actual, raw output from ESLint.**
error  Missing JSDoc comment  require-jsdoc

**Comment by not-an-aardvark** [comment]
It looks like you're using class properties, which are an experimental feature and not generally supported by ESLint's core rules (see [here](https://github.com/eslint/eslint/blob/a8e1c1ccf2a212612f8aa63d7463dfa9acd920b6/README.md#what-about-experimental-features)). You might want to see if [eslint-plugin-babel](https://github.com/babel/eslint-plugin-babel) has a substitute rule for this.
